### PR TITLE
Add SNI documentation

### PR DESCRIPTION
**Pull request by jparisu**
Signed-off-by: jparisu <javierparis@eprosima.com>

### CODE REVIEW

**Comment by jparisu** [comment]
Related with Fast DDS PR: https://github.com/eProsima/Fast-DDS/pull/2950

File: code/XMLTester.xml [hunk 1]
```diff
@@ -610,6 +610,7 @@
                     <option>SINGLE_DH_USE</option>
                     <option>NO_SSLV2</option>
                 </options>
+                <server_name>my_server.com</server_name>
```
Comment (EduPonz) [inline code/XMLTester.xml:613]: This should be added to the corresponding C++ snippet from DDSCodeTester.cpp

**Review by EduPonz — CHANGES_REQUESTED** [review]
The table and XML snippet in section `11.5.1. TLS Configuration` should include the `server_name`

**Review by EduPonz — APPROVED** [review]
LGTM

**Comment by EduPonz** [comment]
@richiprosima please test docs

**Pull request merged**